### PR TITLE
docs: add proposal draft version 1.0

### PR DESCRIPTION
**Pull request by yogeshkathayat**
### Description

added proposal draft version 1.0

### Type of change
- [x] This change adds a documentation

### Checklist:

- [x] Self-review of my own code
- [x] Commented the code, particularly in hard-to-understand areas
- [x] Corresponding changes to the documentation
- [x] Ran test on local
- [x] Generated build on local

### CODE REVIEW

File: PROPOSAL-DRAFT-V1.0.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+A comment can contain any Unicode code point in SourceCharacter except LineTerminator so a comment always consists of all code points starting with the `/*` character up to `*/`
+
+Comments are Ignored like white space and may appear after any token, or before a LineTerminator, and have no significance to the semantic meaning of a TSSG Document.
+```javascript
+/* this is
+multi line 
+comment */
+```
+### Data Types
```
Comment (heypran) [inline PROPOSAL-DRAFT-V1.0.md:115]: Please call this lexical tokens  @tauqeernasir correct me if I am wrong.

File: PROPOSAL-DRAFT-V1.0.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+
+
+## Language
+A TSSG document is defined as a syntactic grammar where terminal symbols are tokens (indivisible lexical units). These tokens are defined in a lexical grammar which matches patterns of source characters.
+This sequence of lexical tokens are then scanned from left to right to produce an abstract syntax tree (AST) according to the Document syntactical grammar.
+
+
+We refer to A TSSG documents as programs. A program may contain expression blocks(schemas, requestbodies, paths, parameters), data types and Ignored lexical grammars(comments, whitespaces, line terminators).
+
```
Comment (heypran) [inline PROPOSAL-DRAFT-V1.0.md:95]: `data types` to lexical tokens

File: PROPOSAL-DRAFT-V1.0.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
+# TSSG
+*Current Working Draft*
+
+## Introduction
+This is the specification for TSSG(The Swagger Schema Generator), which is a new and easy to understand Syntax/Grammar to write OpenAPI Schema in an easy, clean and concise way.
```
Comment (tauqeernasir) [inline PROPOSAL-DRAFT-V1.0.md:5]:
```suggestion
This is the specification for TSSG(The Swagger Schema Generator), which is a new and easy to understand Syntax/Grammar to write OpenAPI Schema in an easy, clean and concise way.
```

This is the specification for TSSG(The Swagger Schema Generator), which is a new and easy to understand Syntax/Grammar that helps writing OpenAPI Schema in short, clean and concise way.

File: PROPOSAL-DRAFT-V1.0.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
```
[the remark this answers is not in the document]
Comment (tauqeernasir) [reply PROPOSAL-DRAFT-V1.0.md:6]: Yes, please also add what @heypran suggested.

File: PROPOSAL-DRAFT-V1.0.md (hunk 1 above)
Comment (tauqeernasir) [reply PROPOSAL-DRAFT-V1.0.md:115]: Yes, they can also be call `lexical tokens`, @yogeshkathayat please update this as well.

File: PROPOSAL-DRAFT-V1.0.md (hunk 3 above)
Comment (mingwho) [reply PROPOSAL-DRAFT-V1.0.md:5]: This is the specification for TSSG(The Swagger Schema Generator), which is a new and easy **way** to understand Syntax/Grammar to write OpenAPI Schema in an easy, clean and concise way.

File: PROPOSAL-DRAFT-V1.0.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+    - [RequestBodies Block](#requestbodies-block)
+      - [RequestBody Expression](#requestbody-expression)
+      - [Extendable RequestBody](#extendable-requestbody)
+    - [Parameters Block](#parameters-block)
+    - [Reference](#reference)
+    - [Paths](#paths)
+
+
+## Overview
+Writing OpenAPI Schema can be tiresome and time wasting task if you write a lot of API Documentation. Updating existing Schema can also be cumbersome and confusing especially when project grows to hundreds of APIs. TSSG is here to help you write schema in an easy, clean and concise way. We have proposed a new and easy to understand Syntax/Grammar for this. It allows you to write less and get full OpenAPI Schema without writing and repeating same line again and again.
```
Comment (mingwho) [inline PROPOSAL-DRAFT-V1.0.md:32]: We have proposed a new and easy **way** to understand Syntax/Grammar for this.

File: PROPOSAL-DRAFT-V1.0.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yogeshkathayat) [reply PROPOSAL-DRAFT-V1.0.md:6]: Thanks, @heypran.  will update it.

File: PROPOSAL-DRAFT-V1.0.md (hunk 3 above)
Comment (yogeshkathayat) [reply PROPOSAL-DRAFT-V1.0.md:5]: Thank You @mingwho .  But I think TSSG is not a way to understand syntax/grammar to write OpenApi Schema. it has its own syntax/grammar. so IMO adding **way** will make it more of a tool to understand grammar. @tauqeernasir @heypran  what do you guys say?

Comment (mingwho) [reply PROPOSAL-DRAFT-V1.0.md:5]: Thanks for the comment. I guess we can write it like this. What do you guys think?
```This is the specification for TSSG(The Swagger Schema Generator), which enables developers to generate OpenAPI Schema with an easy and concise syntax/grammar.```

Comment (yogeshkathayat) [reply PROPOSAL-DRAFT-V1.0.md:5]: LGTM

**Pull request merged**